src/views/Package/PackageState.tsx

```diff
@@ -50,6 +76,165 @@ export const usePackageState = () => {
   return state;
 };
 
+export const appendMenuItem = (
+  items: MenuItem[],
+  item: MenuItem
+): MenuItem[] => {
+  const last = items[items.length - 1];
+
+  if (last && last.level < item.level) {
+    last.children = appendMenuItem(last.children, item);
+    return items;
+  }
+  return [...items, item];
+};
+
+export const splitOnHeaders = (md: string, level: number): string[] => {
+  if (!md) {
+    return [];
+  }
+
+  const regex = new RegExp(`(^#{1,${level}}[^#].*)`, "gm");
```

Comment:
What happens if the author uses [setext headings](https://github.github.com/gfm/#setext-headings) to delineate their headers? Does the markdown still rendered (even if the heading is not identified)? Perhaps this could be added as a test case?

It's probably OK if this isn't supported since it only affects user README's, but IMO we should call this out in a comment and in the PR description.

Reply:
yeah after fiddling with this briefly I think it's best handled in a follow up cause it will take some time to get the regexes correct and change the data flow slightly. Adding a note to the function doc string and updating the PR description.